docs/architecture/adr-050-improved-trusted-peering.md

```diff
@@ -0,0 +1,59 @@
+# ADR 50: Improved Trusted Peering
+
+## Changelog
+* 22-10-2019: Initial draft
+
+## Context
+
+When `max_num_inbound_peers` or `max_num_outbound_peers` of a node reached, the node cannot spare more slots to any peer 
+by inbound or outbound. Therefore, after certain period of disconnection, any important peering can be lost indefinitely 
+because all slots consumed by other peers, and the node stop trying to dial the peer anymore.
+
+This is happening because of two reasons, exponential backoff and absence of unconditional peering feature for trusted peers.
+
+
+## Decision
+
+We would like to suggest solving this problem by introducing two parameters in `config.toml`, `wildcard_peer_ids` and 
+`maximum_dial_period`. 
+
+1) `wildcard_peer_ids`
+
+A node operator inputs list of ids of peers which are allowed to be connected by both inbound or outbound regardless of 
+`max_num_inbound_peers` or `max_num_outbound_peers` of user's node reached or not.
+
+2) `maximum_dial_period`
```

Comment:
Shouldn't it be called `persistent_peers_max_dial_period` if it's only applicable to persistent peers?

Reply:
Agree. We will fix the name. Thanks!

Reply:
Done